src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.cs

```diff
@@ -51,7 +52,11 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                 context.SyntaxProvider
 
                 // Find all MethodDeclarationSyntax nodes attributed with RegexGenerator and gather the required information.
-                .CreateSyntaxProvider(IsSyntaxTargetForGeneration, GetSemanticTargetForGeneration)
+                .ForAttributeWithMetadataName(
+                    context,
+                    RegexGeneratorAttributeName,
+                    (n, c) => n is MethodDeclarationSyntax,
```

Comment:
NIT: can we change `n` -> `node` and `c` -> `cancellationToken`?

Reply:
definitely.